src/SwaggerProvider.DesignTime/v2/OperationCompiler.fs

```diff
@@ -256,30 +256,18 @@ type OperationCompiler (schema:SwaggerObject, defCompiler:DefinitionCompiler, ig
             [
                 ProvidedConstructor(
                     [ProvidedParameter("httpClient", typeof<HttpClient>);
-                     ProvidedParameter("options", typeof<JsonSerializerOptions>)],
-                    invokeCode = (fun args -> <@@ () @@>),
+                     ProvidedParameter("options", typeof<JsonSerializerOptions>, optionalValue = (null:JsonSerializerOptions))],
+                    invokeCode = (fun args ->
+                        match args with
+                        | [] -> failwith "Generated constructors should always pass the instance as the first argument!"
+                        | _ -> <@@ () @@>),
                     BaseConstructorCall = fun args -> (baseCtor, args))
                 ProvidedConstructor(
-                    [ProvidedParameter("httpClient", typeof<HttpClient>)],
-                    invokeCode = (fun args -> <@@ () @@>),
-                    BaseConstructorCall = fun args ->
-                        let args' = args @ [ <@@ null @@> ]
-                        (baseCtor, args'))
-                ProvidedConstructor(
-                    [ProvidedParameter("options", typeof<JsonSerializerOptions>)],
-                    invokeCode = (fun args -> <@@ () @@>),
-                    BaseConstructorCall = fun args ->
-                        let httpClient = <@@ RuntimeHelpers.getDefaultHttpClient defaultHost @@>
-                        let args' =
-                            match args with
-                            | [instance; options] -> [instance; httpClient; options]
-                            | _ -> failwithf "unexpected arguments received %A" args
-                        (baseCtor, args'))
-                ProvidedConstructor([],
+                    [ProvidedParameter("options", typeof<JsonSerializerOptions>, optionalValue = (null:JsonSerializerOptions))],
                     invokeCode = (fun args -> <@@ () @@>),
                     BaseConstructorCall = fun args ->
-                        let httpClient = <@@ RuntimeHelpers.getDefaultHttpClient defaultHost @@>
-                        let args' = args @ [ httpClient; <@@ null @@> ]
+                        let httpClient = <@ RuntimeHelpers.getDefaultHttpClient defaultHost @>
+                        let args' = args @ [httpClient]
```

Comment:
This arguments order is incorrect

Reply:
What is the correct one?

Reply:
The one that you revered in your commit [this; httpClient; options] as base class ctor expects

Reply:
still no, you build `[this; options; httpClient; null]` instead of `[this; httpClient; options]`

Reply:
Than the previous one was absolutely correct.
Because `options` is an optional parameter that is present in both constructors.
I don't need to pass anything more.